### DISCUSSION TITLE
Help wanted

### DISCUSSION THREAD

**Discussion by rdmueller** [docToolchain/docToolchain#607] [General]
Hi Everyone!

Thanx for all your ongoing support. 
I am eager to push the new 2.0 vresion out of the door, but I need your help.

How can you help?

- we need people who are willing to test the new mechanics
- we need people who can help fix problems found 
- we need people who write docs - technical writers!
- we need people who create a nice theme for the new static site generator (jBake). HTML, CSS and Javascript Skills
- we need people who even advance things further

What is in for you?

docToolchain is a community driven project. There is a reason why I moved it away from my personal repository.
With your help, you will make it a real community project of which you can be proud.

If you are interested, please reply to this discussion.

**Comment by uniqueck** [comment]
Hi @rdmueller, 

i would like to help develop doctoolchain 2.0.
I can help develop asciidoctor-extensions from the existing functions, so we can use them a bit more independently. ExportPPT and exportExcel already exits as an asciidoctor extension, I think this is a right approach to make doctoolchain more lightweight and flexible. Furthermore, I think this way we can also provide functional parts to other users who may not want / be able to use the whole DocToolChain stack.

**Reply by rdmueller** [reply]
yes, I guess you are right. This aproach would be cleaner.

btw: there is also already a pandoc gradle plugin to the docx and epub converter:
https://github.com/docToolchain/pandocGradlePlugin

As far as I remember, it was quite close to being ready for the first version.

**Reply by rdmueller** [reply]
if you would like, we can do a mob programming session to exchange some thought and that I learn how I can be of help :-)

**Comment by danielkocot** [comment]
Count me in.

**Comment by rdmueller** [comment]
thanx for responding to this call for help!

My current idea to get things started is to set up some mob programming sessions. 
I would be happy if you could send [me](mailto:ralf.d.mueller@gmail.com) suggestions for individual time slots.
As soon as we have fixed a time slot, I will announce it here so that other might join.

If you have additional idea, feel free to mention them.

**Reply by rdmueller** [reply]
Good News! We have two time slots for docToolchain hacking sessions:

* [2nd of August, 13:00 GMT+2](https://www.timeanddate.com/worldclock/fixedtime.html?iso=20210802T1100) => https://meet.jit.si/docToolchain 
* [26th of August, 19:00 GMT+2](https://www.timeanddate.com/worldclock/fixedtime.html?iso=20210826T1700) => https://meet.jit.si/docToolchain

click on the time/date to see it in your timezone

Feel free to join one of these hack-sessions or send me you suggested time and date for another session :-)

**Reply by uniqueck** [reply]
Hi Ralf,

sorry in the past i was a little bit busy. For the second timeslot i'm in holiday with my childs. 
So maybe we have to discuss some solutions here?

**Reply by rdmueller** [reply]
no problem. if you want to do some pair programming with me or discuss things over a vico, just send me a suggestion for a time and date. Otherwise, let's discuss solutions here or in an issue.

**Comment by uniqueck** [comment]
So my thoughts for the extensions are the following:

- exportJiraIssues and exportJIraSprintChangelog -> one jira related asciidoc extensions
- ....

So we have to decide it for all scripts. The benefits are, we can write tests and they are isolated.
The rest of the stuff is in my opinion an arc42 template generator and setup the doc toolchain stuff, or?

**Reply by rdmueller** [reply]
regardings the extensions - there are currently lots of tasks which could be implemented as extensions, but I wasn't smart enough to find out the right way to do it :-)
I think we could transform these tasks which would result in cleaner code. The questions I do have are:

- should we implement them in ruby so that they are available for all asciidoctor users?
- which features have to be implemented in Java? I guess Excel and PPT export will be better handled in java at the moment.
- how hard will it be to modify these plugins? I have the feeling that the current script approach helped a lot of contributors  to extend some tasks.

**Reply by uniqueck** [reply]
I can't support with Ruby, that I would have to teach myself first, but basically that would be the recommended way also from the AsciiDoctor team.
Powerpoint and Excel the extensions are ready, i think we have to polish them and extend some features, but i think we can use these extension out of the box.

For the rest i would like to start with java, if we decide to support also ruby, than we can create an extension based on ruby and wrap these later as asciidoctorj extension.
As an example this is an wrapped ruby extension https://github.com/uniqueck/asciidoctorj-chart. This extension will be moved to asciidoctor organization, if it is approved.

I think for the extension / modify part we have to build these extension flexibel in part of configuration options.

If i would like to start to implement an extension, I create an repository in my account, or? How would we like to track the progress of the extensions? And what extensions make sense?

So i would like to start with an jira extensions with two block macros issues and sprintChangelog.
We can gladly make a pair programming session, but my time slot is mostly above 10 pm.

**Reply by rdmueller** [reply]
hm. a Java-Wrapper for ruby would be nice rJava :-)
But I do agree that java would be fine for those extensions. And I hope I can learn from you how to create these.

You can start in your own repository, but I will also give you access to the doctoolchain org in case you would like to start there.

regarding the timeslot >10 pm is ok if we stick to 1-2 ours :-)

**Reply by uniqueck** [reply]
Hi Ralf, 

if you have time, we can start a little session now.

**Comment by uniqueck** [comment]
So i started yesterday night with the first extension called asciidoctorj-jira-extension. Do anyone have an idea how we can test it against a working jira instance? I thought i can use a free jira cloud instance, but i got errors if i setup an instance for me. 

Ok but for know it's not so important. We can discuss the format of the blockMacroExtension. 

`jiraIssues::projectKey[<optional JQL>, <optional additional customfields>]`

The base endpoint for jira, is provided as a document attribute.

```
:jiraissues-host: http://jira-cloud.com
:jiraissues-username: <username>
:jiraissues-apitoken: <apitoken>
```
To support i8n, we can provided document attributes for the column headings.

```
:jiraissues-priority-caption: Priority
:jiraissues-cusotmfield123-caption: CustomFieldHeader
```
What do you mean? Any other thoughts or ideas?

**Reply by rdmueller** [reply]
Sorry for the late reply, but I was on vacation and will be again till the end of the week.
It is great that you already started.
Revardjng the jkra instance for tests:
As open source project, we should be able to get a free cloud instance.
If this does not work, afaik atlassian provides docker images with trial versions...

**Reply by uniqueck** [reply]
Hi Ralf,

no problem. I created a first draft for asciidoctorj-jira-extension. Have a look at this code base https://github.com/docToolchain/asciidoctorj-jira-extension

Next week we can start a zoom session.

**Comment by uniqueck** [comment]
I have requested permission from Sonatype for us to publish to Maven Central under the groupId org.doctoolchain.
For progress see here https://issues.sonatype.org/browse/OSSRH-73206
